frappe/model/document.py

```diff
@@ -1058,6 +1058,21 @@ def cancel(self):
 		"""Cancel the document. Sets `docstatus` = 2, then saves."""
 		return self._cancel()
 
+	@frappe.whitelist()
+	def discard(self):
+		"""Discard the draft document. Sets `docstatus` = 2 with db_set."""
+		self.check_if_locked()
+		self.set_user_and_timestamp()
```

Comment:
`check_if_latest` should be done here to avoid parallel updates.

Reply:
Doctype transition check might need some exception for this case somehow.

Reply:
@ankush done